src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -76,14 +75,15 @@ public AssemblyScannerResults GetScannableAssemblies()
                 {
                     if (!assembly.IsDynamic)
                     {
-                        ScanAssembly(AssemblyPath(assembly), results, processed);
```

Comment:
`AssemblyPath` tries to get the assemblies file location which we don't need at all for already loaded assemblies